source/fundamentals/builders/filters.txt

```diff
@@ -11,3 +11,684 @@ Filters Builders
+
+      Bson allComparison = all("vendors", "A", "D");
+      List<Document> results = new ArrayList<>();
+      collection.find(allComparison).into(results);
+      for (Document result : results) {
+         System.out.println(result.toJson());
+      }
+   
+   The Bson renders as:
+
+   .. code-block:: none
+      :copyable: false
+   
+      { 
+         "vendor" : { 
+            "$all" :
+               { 
+                  ["A", "D"] 
+               }
+         }
+      }
+
+   The output of the above code snippet should look something like this:
+   
+   .. code-block:: json
+      :copyable: false
+      
+      {"_id": 8, "color": "orange", "qty": 7, "vendor": ["A", "D"]}
+
+.. _elements:
+
+Elements
+--------
+
+The elements operator methods include:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Elements Operator
+     - Selects
+      
+   * - ``exists()``
+     - documents that have the specified field.
+
+   * - ``type()``
+     - Sdocuments if a field is of the specified type.
+
+The following code snippet shows how to use an elements filter:
+
+.. code-block:: java
+   
+   import static com.mongodb.client.model.Filters.*;
+
+   // <MongoCollection setup code here>
+
+   collection.find(<elements filter>);
+
+.. example::
+
+   The following example selects documents that have a ``qty`` field and
+   its value does not equal ``5`` or ``8`` in the paint_purchaces collection:
+
+   .. code-block:: java
+      
+      import static com.mongodb.client.model.Filters.exists;
+      import static com.mongodb.client.model.Filters.nin;
+      import static com.mongodb.client.model.Filters.and;
+      // <MongoCollection setup code here>
+
+      Bson existsComparison = and(exists("qty"), nin("qty", 5, 8));
+      List<Document> results = new ArrayList<>();
+      collection.find(existsComparison).into(results);
+      for (Document result : results) {
+         System.out.println(result.toJson());
+      }
+
+   The Bson renders as:
+
+   .. code-block:: none
+      :copyable: false
+   
+      {
+         "qty" :
+         [
+            { "$exists" : true,
+            { "$nin" : [5, 8] }
+         ]
+      }
+
+   The output of the above code snippet should look something like this:
+   
+   .. code-block:: json
+      :copyable: false
+      
+      {"_id": 2, "color": "purple", "qty": 10, "vendor": ["C", "D"]}
+      {"_id": 4, "color": "white", "qty": 6, "vendor": ["D"]}
+      {"_id": 5, "color": "yellow", "qty": 11, "vendor": ["A", "B"]}
+      {"_id": 8, "color": "orange", "qty": 7, "vendor": ["A", "D"]}
+
+
+
+.. _evaluation:
+
+Evaluation
+----------
+
+An ``evaluation`` filter to match documemts with an expression.
+
+The evaluation operator methods include:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Evaluation Operator
+     - 
+      
+   * - ``mod()``
+     - Performs a modulo operation on the value of a field and selects documents with a specified result.
```

Comment:
You use "matches" in the other descriptions but "selects" here, is there a difference, or should they be the same for consistency?